tests/model/storage/disk/test_disk_model_store.py

```diff
@@ -46,6 +46,30 @@ def test_store_and_retrieve_model(self):
         # Check that they match.
         self.assertEqual(str(model), str(retrieved_model))
 
+    def test_store_and_retrieve_optimized_model(self):
+        hotkey = "hotkey0"
+        model_id = ModelId(
+            namespace="TestPath",
+            name="TestModel",
+            hash="TestHash1",
+            commit="TestCommit",
+        )
+
+        pt_model = get_model()
+
+        model = Model(id=model_id, pt_model=pt_model)
+
+        # Store the model locally.
+        self.disk_store.store_model(hotkey, model)
+
+        # Retrieve the model locally.
+        retrieved_model = self.disk_store.retrieve_model(
+            hotkey, model_id, optimized=True
```

Comment:
Need to run this test on machine with flash attn installed. Requires a graphics card to even install the library it seems.

Reply:
Can you add a comment about the GPU requirement as a comment on the test? Perhaps we should also add an @skip to this test?

Reply:
Alternatively, I wonder if flash attention should be an optional optimization. We can do a try/catch around the flash attention import and only use it if it's available. WDYT?

Reply:
Added a skip with comments on why. Keeping flash attention as required for now.